src/Bundle/DependencyInjection/Driver/Doctrine/DoctrineORMDriver.php

```diff
@@ -68,7 +68,12 @@ protected function addRepository(ContainerBuilder $container, MetadataInterface
 
             $genericEntities[] = $entityClass;
         } else {
-            $definition->setArguments([$managerReference, $this->getClassMetadataDefinition($metadata)]);
+            if (is_a($repositoryClass, ServiceEntityRepository::class, true)) {
+                $definition->setArguments([new Reference('doctrine')]);
```

Comment:
Why did you decide on a new doctrine reference instead of `new Reference($metadata->getServiceId('manager'));`?